Sources/EduIDExpansion/Flows/Activity/ActivityViewController.swift

```diff
@@ -26,16 +26,16 @@ class ActivityViewController: BaseViewController {
     //MARK: - lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
-        
         screenType = .activityLandingScreen
-        viewModel.loadData()
-        
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
         screenType.configureNavigationItem(item: navigationItem, target: self, action: #selector(dismissActivityScreen))
+        DispatchQueue.main.async { [weak self] in
+            guard let self else { return }
```

Comment:
Why does this need to be async?

Reply:
there maybe be a scenario where an existing Tier user has an account but doesn't have an AccessToken, so the app opens the browser to retrieve an AccessToken, then the user is redirected to the app, I noticed the UI didn't update when user came back from the browser. also the loadData() method is an async task.